target/mips/internal.h

```diff
@@ -344,6 +344,20 @@ static inline int mips_vp_active(CPUMIPSState *env)
     return 1;
 }
 
+static inline bool can_access_cp0(CPUMIPSState* env) {
+    if (((env->CP0_Status & (1 << CP0St_CU0)) &&
+        !(env->insn_flags & ISA_MIPS32R6)) ||
+        !(env->hflags & MIPS_HFLAG_KSU)) {
+#ifdef TARGET_CHERI
+        // For CHERI we need to check PCC.perms before enabling access to cp0 features
+        return (env->active_tc.PCC.cr_perms & CAP_ACCESS_SYS_REGS) != 0;
```

Comment:
This might be nicer written as:

```
static inline bool can_access_cp0(CPUMIPSState* env) {
#ifdef TARGET_CHERI
    // For CHERI we need to check PCC.perms before enabling access to cp0 features
    if (!(env->active_tc.PCC.cr_perms & CAP_ACCESS_SYS_REGS))
        return false;
#endif
   // upstream code
}
```